docs/StardustDocs/topics/schemas/schemas.md

```diff
@@ -0,0 +1,140 @@
+[//]: # (title: Data Schemas)
+
+The Kotlin DataFrame library provides typed data access via
+[generation of extension properties](extensionPropertiesApi.md) for the type
+[`DataFrame<T>`](DataFrame.md) (as well as for [`DataRow<T>`](DataRow.md)), where
+`T` is a marker class representing the `DataSchema` of the [`DataFrame`](DataFrame.md).
+
+A *schema* of a [`DataFrame`](DataFrame.md) is a mapping from column names to column types.  
+This data schema can be expressed as a Kotlin class or interface.  
+If the DataFrame is hierarchical — contains a [column group](DataColumn.md#columngroup) or a
+[column of dataframes](DataColumn.md#framecolumn) — the data schema reflects this structure,
+with a separate class representing the schema of each column group or nested `DataFrame`.
+
+For example, consider a simple hierarchical DataFrame from
+<resource src="example.csv"></resource>.
+
+This DataFrame consists of two columns:
+- `name`, which is a `String` column
+- `info`, which is a [column group](DataColumn.md#columngroup) containing two nested [value columns](DataColumn.md#valuecolumn):
+    - `age` of type `Int`
+    - `height` of type `Double`
+
+<table>
+  <thead>
+    <tr>
+      <th>name</th>
+      <th colspan="2">info</th>
+    </tr>
+    <tr>
+      <th></th>
+      <th>age</th>
+      <th>height</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>Alice</td>
+      <td>23</td>
+      <td>175.5</td>
+    </tr>
+    <tr>
+      <td>Bob</td>
+      <td>27</td>
+      <td>160.2</td>
+    </tr>
+  </tbody>
+</table>
+
+The data schema corresponding to this DataFrame can be represented as:
+
+```kotlin
+// Data schema of the "info" column group
+@DataSchema
+data class Info(
+    val age: Int,
+    val height: Float
+)
+
+// Data schema of the entire DataFrame
+@DataSchema
+data class Person(
+    val info: Info,
+    val name: String
+)
+```
+
+[Extension properties](extensionPropertiesApi.md) for `DataFrame<Person>`  
+are generated based on this schema and allow accessing columns
+or using them in operations:
+
+```kotlin
+// Assuming `df` has type `DataFrame<Person>`
+
+// Get "age" column from "info" group
+df.info.age
+
+// Select "name" and "height" columns
+df.select { name and info.height }
+
+// Filter rows by "age"
+df.filter { age >= 18 }
+```
+
+See [](extensionPropertiesApi.md) for more information.
+
+
+## Schemas retrieving
+
+Defining a data schema manually can be difficult, especially for dataframes with many columns or deeply nested structures, and may lead to mistakes in column names or types. Kotlin DataFrame provides several methods for generating data schemas.
+
+* [**`generate..()` methods**](DataSchemaGenerationMethods.md) are extensions for [`DataFrame`](DataFrame.md) that generate a code string representing its `DataSchema`.
```

Comment:
also extensions for `DataFrameSchema`, might be relevant

Reply:
What do you mean?

Reply:
please, show a usecase